lib/action_mailer/logged_smtp_delivery.rb

```diff
@@ -2,6 +2,11 @@
 require 'mail/network/delivery_methods/smtp'
 
 class ActionMailer::LoggedSMTPDelivery < Mail::SMTP
+  # The ULID pattern conforms with the spec - https://github.com/ulid/spec#encoding
+  # a. Canonically encoded as a 26 character string
+  # b. Excludes the letters I, L, O, and U to avoid confusion and abuse.
+  ULID_PATTERN = /[A-HJKMNP-TV-Z0-9]{26}/.freeze
```

Comment:
Just curious if negative lookahead might make this more readable. Something like 
/(?!.*[ILOU])[A-Z0-9]{26}/

Reply:
This looks more readable and clear. Thanks for suggesting it.